netflex-practice/src/Route/Login/index.tsx

```diff
@@ -0,0 +1,111 @@
+import { FC } from 'react';
+import { useForm, SubmitHandler } from 'react-hook-form';
+import { useNavigate } from 'react-router-dom';
+import styles from './style.module.scss';
+type FormData = {
+  userName: string;
+  password: string;
+  rePassword: string;
+  mail: string;
+};
+export const Login: FC = () => {
+  const navigate = useNavigate();
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset,
+    watch,
+  } = useForm<FormData>({
+    mode: 'onChange',
+    defaultValues: {
+      userName: '',
+      password: '',
+      rePassword: '',
+      mail: '',
+    },
+  });
+  const onSubmit: SubmitHandler<FormData> = () => {
+    if (
+      !errors.userName &&
+      !errors.password &&
+      !errors.rePassword &&
+      !errors.mail
+    ) {
+      navigate('/Home');
+      reset();
+    }
+  };
+  const password = watch('password', 'default value');
+  return (
+    <form onSubmit={handleSubmit(onSubmit)}>
+      <h1>ログイン</h1>
+      <div className={styles.Login}>
+        <label className={styles.LoginLabel}>userName</label>
+        <input
+          type='text'
+          className={styles.LoginInput}
+          {...register('userName', {
+            required: 'ユーザー名は必須です',
+            pattern: {
+              value: /^([a-zA-Z0-9]{8,15})$/,
+              message:
+                'ユーザー名は半角英数字8文字以上、15文字以下で入力してください',
+            },
+          })}
+        ></input>
```

Comment:
[nits]
子要素がなければ不要かも

```suggestion
    <input
          type='text'
          className={styles.LoginInput}
          {...register('userName', {
            required: 'ユーザー名は必須です',
            pattern: {
              value: /^([a-zA-Z0-9]{8,15})$/,
              message:
                'ユーザー名は半角英数字8文字以上、15文字以下で入力してください',
            },
          })}
        / >
```

Reply:
inputタグ全て上記の様になっていたので、修正しました。